sig-scalability/slos/network_latency.md

```diff
@@ -0,0 +1,47 @@
+## In-cluster network latency SLIs/SLOs details
+
+### Definition
+
+| Status | SLI | SLO |
+| --- | --- | --- |
+| __WIP__ | In-cluster network latency from a single node, measured as latency of per second ping to kubernetes.default.svc.cluster.local/, measured as 99th percentile over last 5 minutes. | In default Kubernetes installataion with RTT between nodes <= Y, 99th percentile of (99th percentile over all nodes) per cluster-day <= X |
```

Comment:
Should we use the master for this SLI? It is "special" in some sense, as in some implementations, the master lives on a different networking plane.

Reply:
Also, ping from where. And you probably mean TCP RTT, not ICMP.

Reply:
(Also, we should probably use a non-TLS-based endpoint to eliminate that latency)

Reply:
Does it make sense to say "RTT to a null service" where null service is defined to be some hello-world serving container?

Reply:
> Also, we should probably use a non-TLS-based endpoint to eliminate that latency

Yes - that absolutely makes sense.

> Does it make sense to say "RTT to a null service" where null service is defined to be some hello-world serving container?

Yes - that would be the ideal goal. Though, what I wanted to avoid is to create a dedicated service for that (or at least, creating a dedicated pods to serve those service), because we would like to measure that in user clusters too, and we don't want to consume additional user resources.

Maybe the workaround for it would be to make this "null service" be exposed from kube-proxy pods and then we would just need to create a service (which is not that big deal).
The drawback of that is that then there is a backend on every single node, so this won't work good in large clusters.
[Ideally, i would say let's expose it from a subset of kube-proxies, but then we would need to add labels on some kube-proxies, which then turns out to be too complicated, i think...]

Maybe you have something better on your mind?

Reply:
As mentioned above we don't seem to have anything satisfying our needs now.
Making a service on top of kube-proxies will be problematic in large clusters, and choosing only a subset complicates cluster setup.

So we decided for requiring that null-service to be setup by cluster administrator.